add image build(s)

PeerHerholz opened the pull request:
- file generation through neurodocker
- docker and/or singularity
- option to build image(s) locally

PeerHerholz committed:
add image build(s)

- include options to build images through neurodocker
- includes docker and singularity

PeerHerholz commented:
@JoseAlanis, I just remembered that we need FreeSurfer for source reconstruction, etc., eh?

JoseAlanis commented on Dockerfile:
```diff
@@ -0,0 +1,138 @@
+           'rpy2=2.9.4' \
+           'r-lme4' \
+           'r-lmertest' \
+           'r-lsmeans' \
```
lets go with ``emmeans`` instead of ``lsmeans``. I think their going to archive ``lsmeans`` in the near future.

PeerHerholz replied:
kk, thx for the pointer. I will update this part.

PeerHerholz replied:
Done [here](https://github.com/C0C0AN/Mindboggle-MixedModels/pull/2/commits/e63eba433083f2566dae2b717b4c0a1ee201ef64).

JoseAlanis commented:
> @JoseAlanis, I just remembered that we need FreeSurfer for source reconstruction, etc., eh?

well it depends I guess, do we wanna do source reconstruction here?

 I'll try to look for some other insteresting packages today (e.g., for viz).

JoseAlanis reviewed (commented):
Do we wanna use ``ggplot2`` for vizualisation? We have some code for that alredy, right? What do you think, perhaps we could just reuse / imporve it?

PeerHerholz commented:
> > @JoseAlanis, I just remembered that we need FreeSurfer for source reconstruction, etc., eh?
> 
> well it depends I guess, do we wanna do source reconstruction here?

Oh sorry, I totally mixed up the repos. I meant to post that in the [EEGprep docker PR](https://github.com/C0C0AN/EEGprep/pull/2). Never mind, we don't need FreeSurfer here.

PeerHerholz commented:
> Do we wanna use `ggplot2` for vizualisation? We have some code for that alredy, right? What do you think, perhaps we could just reuse / imporve it?

Yes, this seems about right. The idea would be to generate informative `.html` files, that include basic statistics about the values (mindboggle output), that is an interactive version of the `.csv` file, distribution of the data, some tests (e.g., normality, etc.) and the outcomes of the mixed models as static and interactive graphic. The `.html` file will be based on a template and the add the respective information/graphics after the BIDSapp was run, with the corresponding files (e.g., summary `.csv`, model outcomes, etc.) saved as files themselves in the same folder. Do you have any idea if `ggplot2` works as `plotly` in that if embedded in a `.html` remains interactive even if the corresponding graphic is not shared along with the `.html`? It should if we directly include the produced `ggplot2` `.html` code, eh?

PeerHerholz committed:
update image generation

- include r-emmeans instead of r-lsmeans as the later might be archived in the near future

PeerHerholz committed:
reorder files & file uploads

- codecov.yml, .coveragerc, .flake8,. gitattributes, .gitignore and .travis.yml were moved from mindboggle-mixedmodels to main repo folder
- introduction of basic function structure into workflows, utils and test
- include initial draft of report_template.html
- introduce gen_analyses_config_json.py
